src/TreeSitter/Importing.hs

```diff
@@ -128,7 +158,56 @@ instance Monad m => Monad (MaybeC m) where
     a' <- a
     case a' of
       Nothing -> pure Nothing
-      Just a -> runMaybeC $ f a
+      Just a  -> runMaybeC $ f a
 
 instance MonadIO m => MonadIO (MaybeC m) where
   liftIO = MaybeC . fmap Just . liftIO
+
+
+newtype FieldName = FieldName { getFieldName :: String }
```

Comment:
Examples of field names are left, right, operator, or for an if statement, something like `condition`, `consequence`, `alternative`. In the `CodeGen.Deserialize` code, field name is used to name the record fields of product data types. 

The generic building code uses the field name for the current node to select which record field should build with the sub-tree.